trtools/utils/tr_harmonizer.py

```diff
@@ -1141,10 +1147,19 @@ def GetDosages(self,
 
             # Check AP field. allow wiggle room for rounding
             if np.any(np.sum(ap1, axis=1) > 1.1) or np.any(np.sum(ap2, axis=1) > 1.1):
-                raise ValueError("AP1 or AP2 field summing to more than 1 detected")
+                error_msg = "{}:{} AP1 or AP2 field summing to more than 1 detected".format(self.chrom, self.pos)
+                if strict:
+                    raise ValueError(error_msg)
+                else:
+                    common.WARNING(error_msg)
+                    return np.array([np.nan]*self.GetNumSamples())
             if np.any(ap1 < 0) or np.any(ap2 < 0):
-                raise ValueError("Negative AP1 or AP2 fields detected")
-
+                error_msg = "{}:{} Negative AP1 or AP2 fields detected".format(self.chrom, self.pos)
+                if strict:
+                    raise ValueError("Negative AP1 or AP2 fields detected")
+                else:
+                    common.WARNING(error_msg)
+                    return np.array([np.nan]*self.GetNumSamples())
```

Comment:
Same comment as above. Did you make an explicit decision to output NaN for all samples instead of just the ones that failed this check?
```suggestion
            failing = (ap1 < 0) | (ap2 < 0)
            if np.any(failing):
                error_msg = "{}:{} Negative AP1 or AP2 fields detected".format(self.chrom, self.pos)
                if strict:
                    raise ValueError("Negative AP1 or AP2 fields detected")
                else:
                    common.WARNING(error_msg)
                    ap1[failing,:] = np.nan
                    ap2[failing,:] = np.nan
```